rs/canister_sandbox/src/replica_controller/sandboxed_execution_controller.rs

```diff
@@ -2645,4 +2665,148 @@ mod tests {
         let metric = &m.sandboxed_execution_subprocess_memfd_rss_total;
         assert_eq!(metric.get(), 0); // no memfd.
     }
+
+    fn api_memory_key(api_type: &str, memory_type: &str) -> BTreeMap<String, String> {
+        BTreeMap::from([
+            ("api_type".into(), api_type.into()),
+            ("memory_type".into(), memory_type.into()),
+        ])
+    }
+
+    #[rstest]
+    #[case::canister_does_not_trap("", ErrorCode::CanisterDidNotReply)]
+    #[case::canister_traps("(unreachable)", ErrorCode::CanisterTrapped)]
+    fn sigsegv_handler_duration_metric_is_reported(
+        #[case] inject_trap: &str,
+        #[case] expected_error_code: ErrorCode,
+    ) {
+        let mut test = ExecutionTestBuilder::new().build();
+        let wat = format!(
+            r#"
+            (module
+                (import "ic0" "stable64_write"
+                    (func $stable_write (param $offset i64) (param $src i64) (param $size i64))
+                )
+                (import "ic0" "stable64_grow" (func $stable_grow (param i64) (result i64)))
+                (func (export "canister_update write_heap")
+                    (i32.store (i32.const 0) (i32.const 42))
+                    {inject_trap}
+                )
+                (func (export "canister_update write_stable")
+                    (drop (call $stable_grow (i64.const 1)))
+                    (call $stable_write (i64.const 0) (i64.const 0) (i64.const 1))
+                    {inject_trap}
+                )
+                (memory 1)
+            )"#
+        );
+        let canister_id = test.canister_from_wat(wat).unwrap();
+
+        let err = test.ingress(canister_id, "write_heap", vec![]).unwrap_err();
+        assert_eq!(err.code(), expected_error_code);
+        let metrics = fetch_histogram_vec_stats(
+            test.metrics_registry(),
+            "sandboxed_execution_sigsegv_handler_duration_seconds",
+        );
+
+        let value = metrics.get(&api_memory_key("update", "wasm")).unwrap();
+        assert_eq!(value.count, 1);
+        assert!(value.sum > 0.000001); // 1 µs
```

Comment:
nit: Maybe define a constant for the "magic" value of 1μs. Similar for the test below and the other value.